static/app/utils/project/useProject.tsx

```diff
@@ -0,0 +1,80 @@
+import {useCallback, useEffect, useMemo} from 'react';
+
+import type {ApiResult} from 'sentry/api';
+import type {Project} from 'sentry/types';
+import useAggregatedQueryKeys from 'sentry/utils/api/useAggregatedQueryKeys';
+import useAllProjectVisibility from 'sentry/utils/project/useAllProjectVisibility';
+import type {ApiQueryKey} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type Props =
+  | {slug: string | undefined; id?: undefined}
+  | {id: string | undefined; slug?: undefined};
```

Comment:
i know you're avoiding checking if the object contains the keys, could use never too
```suggestion
  | {slug: string | undefined; id?: never}
  | {id: string | undefined; slug?: never};
```